### PR TITLE
Improve anti-spam logs

### LINKED ISSUE #1850: Always create a log if filtered messages have attachments.
Currently, when creating a mod alert, the bot will not create a log of the messages if there is only one message involved, and will display it in the alert embed directly instead.

That should not be the case when there are attachements. If there are any, it should always create a new log and link to it instead of trying to display the contents of the message (which might not exist).

Example of an alert with no content (because it's a single message with only attachments):
image

How it looks like when it's several messages:
image

### PR DESCRIPTION
**Pull request by TizzySaurus**
Closes #1850.

Messages triggering antispam which have attachments will now *always* be uploaded to the log site, and messages that are truncated will now be uploaded to the paste site as well as showing the truncated version of the message.

### CODE REVIEW

File: bot/exts/filters/antispam.py [hunk 1]
```diff
@@ -83,7 +83,11 @@ async def upload_messages(self, actor_id: int, modlog: ModLog) -> None:
         )
 
         # For multiple messages or those with excessive newlines, use the logs API
-        if len(self.messages) > 1 or 'newlines' in self.rules:
+        if any((
+            len(self.messages) > 1,
+            self.messages[0].attachments,
```
Comment (Akarys42) [inline bot/exts/filters/antispam.py:88]:
```suggestion
            len(self.messages[0].attachments) > 0,
```

**Review by Akarys42 — CHANGES_REQUESTED** [review]
I'd prefer if that check would explicitly use `len()`

File: bot/exts/filters/antispam.py (hunk 1 above)
Comment (TizzySaurus) [reply bot/exts/filters/antispam.py]: Fair enough, have committed :+1:

File: bot/exts/filters/antispam.py [hunk 2]
```diff
@@ -83,7 +83,11 @@ async def upload_messages(self, actor_id: int, modlog: ModLog) -> None:
         )
 
         # For multiple messages or those with excessive newlines, use the logs API
-        if len(self.messages) > 1 or 'newlines' in self.rules:
+        if any((
+            len(self.messages) > 1,
+            len(self.messages[0].attachments) > 0,
+            self.messages[0].count('\n') > 15
+        )):
```
Comment (wookie184) [inline bot/exts/filters/antispam.py:90]: `self.messages` is a dictionary mapping message IDs to message objects, so this doesn't work.
I get this error:
```
bot_1        | 2021-10-05 12:13:33 | bot.utils.scheduling | ERROR | Error in task AntiSpam._process_deletion_context(frozenset({<Member id=493839819168808962 name='wookie' discriminator='4017' bot=False nick=None guild=<Guild id=813116921646022716 name="wookie's test server" shard_id=None chunked=True member_count=4>>})) 139705866738800!
bot_1        | Traceback (most recent call last):
bot_1        |   File "/bot/bot/exts/filters/antispam.py", line 299, in _process_deletion_context
bot_1        |     await deletion_context.upload_messages(self.bot.user.id, self.mod_log)
bot_1        |   File "/bot/bot/exts/filters/antispam.py", line 89, in upload_messages
bot_1        |     len(self.messages[0].attachments) > 0,
bot_1        | KeyError: 0
```

**Review by wookie184 — CHANGES_REQUESTED** [review]

**Comment by wookie184** [comment]
looks good other than the linting error

**Review by wookie184 — APPROVED** [review]
LGTM

**Review by Akarys42 — APPROVED** [review]
Lgtm

**Pull request merged**